architecture/src/androidLibMain/kotlin/viewmodel/KalugaViewModelLifecycleObserver.kt

```diff
@@ -39,12 +37,12 @@ import kotlin.reflect.full.starProjectedType
  */
 class KalugaViewModelLifecycleObserver<VM : BaseViewModel> internal constructor(private val viewModel: VM, private val activity: Activity?, private val lifecycleOwner: LifecycleOwner, private val fragmentManager: FragmentManager) : LifecycleObserver {
 
-    private val publicVmProperties: List<KProperty1<VM, Any?>> by lazy {
+    private val lifecycleSubscribables: List<LifecycleSubscribable> by lazy {
         viewModel::class.memberProperties
             .mapNotNull { it as? KProperty1<VM, Any?> }
-            .filter { it.getter.visibility == KVisibility.PUBLIC && it.getter.returnType.isSubtypeOf(LifecycleSubscribable::class.starProjectedType) }
+            .filter { it.getter.visibility == KVisibility.PUBLIC && it.getter(viewModel) is LifecycleSubscribable }
```

Comment:
Since this it now already getting the property directly, I think this should no longer be using lazy, but rather run on every bind, since it now needs to account for variable properties

Reply:
Yes, that's a valid point, but I'd rather take a slightly different approach. I'm not sure `LifecycleSubscribable` is supposed to be a mutable property, so I'd suggest to explicitly check, that it is not a `KMutableProperty1`. Otherwise we need to take care of other cases, like when the property is updated, we should "unbind" an old instance. What do you think?

Reply:
Yeah agreed. It should be immutable preferably